decidim-collaborative_texts/app/controllers/decidim/collaborative_texts/documents_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Decidim
+  module CollaborativeTexts
+    # Exposes the blog resource so users can view them
+    class DocumentsController < Decidim::CollaborativeTexts::ApplicationController
+      include Decidim::Paginable
+
+      helper_method :documents, :document, :paginate_documents
+
+      def index; end
+
+      def show
+        raise ActionController::RoutingError, "Not Found" unless document
+      end
+
+      private
+
+      def document
+        @document ||= documents.find(params[:id])
+      end
+
+      def documents
+        @documents ||= if current_user&.admin?
+                         Document.where(component: current_component)
+                       else
+                         Document.published.where(component: current_component)
+                       end
+      end
+
+      def paginate_documents
+        @paginate_documents ||= paginate(documents.published.enabled_desc)
```

Comment:
You may have a bug here. If the user is admin, you are ignored published state in the main finder. 

```suggestion
        @paginate_documents ||= paginate(documents.enabled_desc)
```

Reply:
Done